posthog/clickhouse/query_fragment.py

```diff
@@ -0,0 +1,99 @@
+import itertools
+import re
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Dict, List, Mapping, Sequence, Tuple, Union
+
+ParamValueType = Union[str, int, float, None, List]
+QueryFragmentLike = Union["QueryFragment", str]
+
+unique_sequence = itertools.count()
+
+
+@dataclass(frozen=True)
+class Param:
+    value: ParamValueType
+
+
+@dataclass
+class QueryFragment:
```

Comment:
I 💖 that this gives a name to something that is repeated across tens (if not hundreds) of files